src/GoogleFit/FitOnFhir.GoogleFit/Client/Handlers/GoogleFitAuthorizationHandler.cs

```diff
@@ -45,60 +47,68 @@ private GoogleFitAuthorizationHandler()
             _tokenValidationService = EnsureArg.IsNotNull(tokenValidationService, nameof(tokenValidationService));
             _authStateService = EnsureArg.IsNotNull(authStateService, nameof(authStateService));
             _logger = EnsureArg.IsNotNull(logger, nameof(logger));
+            InitHandledRoutes();
         }
 
         public static IResponsibilityHandler<RoutingRequest, Task<IActionResult>> Instance { get; } = new GoogleFitAuthorizationHandler();
 
-        public async Task<IActionResult> Evaluate(RoutingRequest request)
+        public override IEnumerable<string> HandledRoutes => _handledRoutes;
+
+        protected Func<string, bool> IsRouteHandled => (x) => { return HandledRoutes.Any(str => str == x); };
+
+        public override Task<IActionResult> Evaluate(RoutingRequest request)
         {
-            try
+            var route = EnsureArg.IsNotNullOrWhiteSpace(request.HttpRequest.Path.Value?[1..]);
+
+            if (!IsRouteHandled(route))
             {
-                var path = EnsureArg.IsNotNullOrWhiteSpace(request.HttpRequest.Path.Value?[1..]);
+                return null;
+            }
 
-                if (path.StartsWith(GoogleFitConstants.GoogleFitCallbackRequest))
-                {
-                    return await Callback(request);
-                }
+            if (route.StartsWith(GoogleFitConstants.GoogleFitCallbackRequest))
+            {
+                return Callback(request);
+            }
+
+            return HandleAuthorizeRequest(request, route);
+        }
 
-                AuthState state;
-                var isValidated = await _tokenValidationService.ValidateToken(request.HttpRequest, request.Token);
-                if (isValidated)
+        protected async Task<IActionResult> HandleAuthorizeRequest(RoutingRequest request, string route)
+        {
+            AuthState state;
+            var isValidated = await _tokenValidationService.ValidateToken(request.HttpRequest, request.Token);
+            if (isValidated)
+            {
+                try
                 {
-                    try
-                    {
-                        state = _authStateService.CreateAuthState(request.HttpRequest);
-                    }
-                    catch (ArgumentException)
-                    {
-                        return new BadRequestObjectResult($"'{Constants.ExternalIdQueryParameter}' and '{Constants.ExternalSystemQueryParameter}' are required query parameters.");
-                    }
-                    catch (RedirectUrlException ex)
-                    {
-                        return new BadRequestObjectResult(ex.Message);
-                    }
+                    state = _authStateService.CreateAuthState(request.HttpRequest);
                 }
-                else
+                catch (ArgumentException)
```

Comment:
These try/catch blocks have a bug. In the `_authStateService`, we throw an `ArgumentException` that is caused by the exact opposite condition. This will catch the `ArgumentException` and write the exact opposite message.

I think we need to handle this differently. Maybe we have a type AuthStateException that can be thrown for 400 Bad Request conditions and all other exceptions will be caught by the "global" RoutingService try/catch.